### PR TITLE
Fix browser.test_async_2

### PR DESCRIPTION
**Pull request by kripken**
The test regressed after an LLVM change. The LLVM change removes the
recursion (!), which then lets binaryen inline it, which changes the stack
traces radically. To work around that, do not run the binaryen optimizer.

I looked at the history of the test to make sure the optimizer wasn't a part
of what is tested, and it seems not. The test originates in the emterpreter,
so it was not even an asyncify test originally. But even there, the test checked
for trampling of locals due to the async transform, and that would be the
same in an opt build as otherwise (that is, the same local storing/loading code
is used in all cases; if the optimizer breaks that, it would be a general optimizer
bug and not specific to asyncify).

Also mark the function as noinline, which is something the test depends on
heavily.

### CODE REVIEW

File: tests/browser/async_2.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 #include <emscripten.h>
 #include <assert.h>
 
-int calc(int x) {
+__attribute__((noinline)) int calc(int x) {
```
Comment (sbc100) [inline tests/browser/async_2.cpp:10]: Is this line alone not enough to fix the issue?

**Review by sbc100 — APPROVED** [review]

File: tests/browser/async_2.cpp (hunk 1 above)
Comment (kripken) [reply tests/browser/async_2.cpp:10]: No, since it doesn't apply to binaryen sadly...

Comment (kripken) [reply tests/browser/async_2.cpp:10]: (but this might prevent future regressions if LLVM wants to try to inline it...)

**Comment by sbc100** [comment]
BTW this test looks on the face of it rather over complicated.  Does it somehow rely on all those different recursive paths?  It kind of looks like like it might be trying to confound an optimizer but that seems like sketchy thing to be relying on.

**Comment by kripken** [comment]
Yes, i think the recursion is an attempt to prevent inlining and tail call opts etc. Seems like LLVM got too smart for it...

It's possible the test can just be removed. I'm not sure what it tests today that is not otherwise covered. However, we'd need to spend more time to verify that I think.

**Pull request merged**